delfin/drivers/ibm/storwize_svc/ssh_handler.py

```diff
@@ -0,0 +1,363 @@
+                    'storage_id': self.storage_id,
+                    'description': '',
+                    'status': status,
+                    'native_volume_id': str(map.get('id')),
+                    'native_storage_pool_id': map.get('mdisk_grp_id'),
+                    'wwn': str(map.get('vdisk_UID')),
+                    'type': volume_type,
+                    'total_capacity': int(total_capacity),
+                    'used_capacity': int(used_capacity),
+                    'free_capacity': int(free_capacity),
+                    'compressed': compressed,
+                    'deduplicated': deduplicated
+                }
+                volume_list.append(v)
+
+            return volume_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get storage volume: %s" % (e.msg)
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get storage volume: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def judge_alert_time(self, map, query_para):
+        if len(map) <= 1:
+            return False
+        if query_para is None and len(map) > 1:
+            return True
+        time_array = time.strptime(map.get('last_timestamp'),
+                                   SSH_Handler.TIME_PATTERN)
+        occur_time = int(time.mktime(time_array) * 1000)
+        if query_para.get('begin_time') and query_para.get('end_time'):
+            if occur_time >= int(query_para.get('begin_time')) and \
+                    occur_time <= int(query_para.get('end_time')):
+                return True
+        if query_para.get('begin_time'):
+            if occur_time >= int(query_para.get('begin_time')):
+                return True
+        if query_para.get('end_time'):
+            if occur_time <= int(query_para.get('end_time')):
+                return True
+        return False
+
+    def list_alerts(self, context, query_para):
+        try:
+            alert_list = []
+            alert_info = self.handle_detail_ssh(
+                context, self.eventlog_command_without_filter, 'alert')
+            alert_res = alert_info.split('\n')
+            for i in range(1, len(alert_res)):
+                if alert_res[i] is None or alert_res[i] == '':
+                    continue
+                alert_str = ' '.join(alert_res[i].split())
+                strinfo = alert_str.split(' ', 9)
+                detail_command = \
+                    SSH_Handler.alert_detail_command + strinfo[0]
+                deltail_info = self.handle_detail_ssh(
+                    context, detail_command, 'alertdetail')
+                map = {}
+                self.handle_detail(deltail_info, map, split=' ')
+                if self.judge_alert_time(map, query_para) is False:
+                    continue
+                time_str = '20%s' % map.get('last_timestamp')
```

Comment:
output also contains 'last_timestamp_epoch' .I think this helps better for conversion

Reply:
I have changed this issue